processing/src/main/java/io/druid/query/metadata/SegmentMetadataQueryQueryToolChest.java

```diff
@@ -107,12 +108,14 @@ public SegmentMetadataQueryQueryToolChest(SegmentMetadataQueryConfig config, Gen
           Map<String, Object> context
       )
       {
-        Query<SegmentAnalysis> query = queryPlus.getQuery();
+        SegmentMetadataQuery castedQuery = (SegmentMetadataQuery) queryPlus.getQuery();
+        SegmentMetadataQuery updatedQuery =(SegmentMetadataQuery) (castedQuery.withAnalysisTypes(getFinalAnalysisTypes(castedQuery)));
```

Comment:
`withAnalysisTypes()` could return `SegmentMetadataQuery` and casting not needed.